### PR TITLE
[SCRUM-191] The email history spreadsheet file name causes an email history error

### PR DESCRIPTION
**Pull request by sh1un**
## Description

Fixed two issues:
1. Added null check handling for spreadsheet file name in email history list
2. Improved UI for better UX:
   - Enhanced recipient source display in email history list
   - Updated title and description in email details page
## Type of Change

- [x] 🐛 Bug Fix（修復 bug）
- [ ] ✨ New Feature（新功能）
- [x] 💄 UI/UX（介面或使用者體驗優化）
- [x] 🔨 Refactor（程式碼重構）
- [ ] 📝 Documentation（文件更新）
- [ ] 🔧 Configuration（配置變更）
- [ ] 🚀 Performance（效能優化）
- [ ] ✅ Test（測試相關）
- [ ] 🔒 Security（安全性相關）

## Related Issues

[SCRUM-191](https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-191)

## Testing

- You can go to the [preview environment](https://email-sender-hgdim0m9l-aws-educate-projects.vercel.app/emailHistory) to review the new UI

## Screenshots/Videos

1. emailHistory UI improvements
  - before
    image
  - after
    image
2. emailHistory/[runId] UI improvements
  - before
    image
  - after
    image

## Checklist

- [x] 我已經測試過這些變更
- [ ] 我已經更新相關文件
- [x] 我的程式碼遵循專案的程式碼規範
- [x] 我已經處理所有的 console warnings/errors
- [ ] 我已經移除所有的 console.log
- [ ] 我已經新增必要的測試案例
- [x] 所有現有的測試都能通過

## Additional Notes

N/A

[SCRUM-191]: https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-191?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/emailHistory/[runId]/page.tsx [hunk 1]
```diff
@@ -92,10 +91,10 @@ export default function Page({ params }: PageProps) {
   return (
     <>
       <div className="flex flex-col justify-center items-start">
-        <p className="text-4xl font-bold pt-2">Emails history</p>
+        <p className="text-4xl font-bold pt-2">Email Sending Details</p>
         <div className="flex justify-between items-center w-full pb-4">
           <p className="text-gray-500 italic">
-            Details of <strong>one of the runs </strong>is shown here.
+            Detailed delivery status for run ID: <strong>{params.runId}</strong>
```
Comment (chungchihhan) [inline src/app/emailHistory/[runId]/page.tsx:97]: Is it neccessary to tell the user about the run ID?

File: src/app/ui/email-history-card.tsx [hunk 2]
```diff
@@ -42,110 +18,163 @@ interface senderType {
...
                   </div>
                   <strong>{convertToTaipeiTime(item.created_at)}</strong>
                 </div>
+
                 <div className="flex items-center gap-2">
                   <div className="flex items-center gap-2">
                     <Send size={20} />
                     <p>Sender</p>
                   </div>
                   <strong>{item.sender.username}</strong>
```
Comment (chungchihhan) [inline src/app/ui/email-history-card.tsx:163]: Deal with the empty Sender 

image
image

File: src/app/ui/email-history-card.tsx [hunk 3]
```diff
@@ -42,110 +18,163 @@ interface senderType {
...
   created_year: string;
 }
 
-interface props {
+interface propsType {
   data: dataType[] | null;
 }
 
-export default function EmailHistoryCard({ data }: props) {
+export default function EmailHistoryCard({ data }: propsType) {
```
Comment (chungchihhan) [inline src/app/ui/email-history-card.tsx:55]: Better to return for empty or undefined data before `renderRecipientInfo`
```
  if (!data || data.length === 0) {
    return <div className="w-full p-8 text-center text-gray-500">No email history found</div>;
  }
```

**Comment by chungchihhan** [comment]
The [preview](https://email-sender-hgdim0m9l-aws-educate-projects.vercel.app/emailHistory) here needs to login first. So try to navigate to the login page to login first. 

We got [SCRUM-193](https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-193?atlOrigin=eyJpIjoiNjE4M2MyYzE2NzU5NGM3Mzg3Njg3MmJlMDg4MjYyMDIiLCJwIjoiaiJ9) to deal with this kind of login problem. So we will add this check in SCRUM-193

**Comment by sh1un** [comment]
Hi @chungchihhan 

I've addressed all the feedback and pushed the changes. Please review agiain when you have time.


#### UI updates

1. Null sender ([comment link](https://github.com/aws-educate-tw/email-sender-nextjs/pull/50#discussion_r1899366555))
    image

2. Removed run ID ([comment link](https://github.com/aws-educate-tw/email-sender-nextjs/pull/50#discussion_r1899353684))
    image

**Review by chungchihhan — APPROVED** [review]
# Code Review Checklist

## 基礎檢查
- [x] 確認自動化測試全部通過
- [x] 程式碼沒有合併衝突
- [x] PR描述清楚且完整

## 主要檢查項目
### 程式設計
- [x] 功能實作是否正確
- [x] 有適當的錯誤處理
- [x] 實作複雜度是否合理

### 程式品質
- [x] 變數/函數命名是否清晰
- [x] 程式碼容易理解
- [x] 符合專案程式風格

### 測試
- [x] 新功能有適當的測試
- [x] 已測試重要的邊界條件

### 文件
- [ ] 必要的文件是否更新
- [ ] API文件是否同步更新

## Review 建議事項：
N/A

## 整體評價
- [x] LGTM (Looks Good To Me) 
- [ ] 需要修改後再審查

**Pull request merged**